cmake_modules/ParsecCompilerFlags.cmake

```diff
@@ -128,15 +128,15 @@ endif(NOT CMAKE_SYSTEM_NAME MATCHES "Windows")
 #
 
 # add gdb symbols in debug and relwithdebinfo, g3 for macro support when available
-check_and_set_compiler_option(OPTION "-g3" NAME PARSEC_HAVE_G3 COMMENT "List of compile options used to enable highest level of debugging (e.g. -g3)")
+check_and_set_compiler_option(OPTION "-g3" NAME PARSEC_HAVE_G3 COMMENT "List of compile options used to enable highest level of debugging (e.g. -g3)" CONFIG DEBUG)
```

Comment:
Why only enable `-g3` for debug builds? I don't think this is harmful for RelWithDebInfo builds

Reply:
I knew somebody would ask :) the real reason is that the `check_and_set_compiler_option` doesn't take a list of CONFIG in the parameter, but a single one, and didn't felt like doing more cmake today.

Reply:
OK, I can live with that. This explains why I had pretty decent debug symbols in my `Release` builds. We might want to keep that after all to make debugging even in a release build.

Reply:
We still get normal -g in RelWithDebInfo, and we definitely should not get -g in Release (that is the difference between RelWithDebInfo and Release, after all)

Reply:
Yes and no. RelWithDebInfo disables most optimizations. So bugs appear in Release mode and not in RelWithDebInfo.